samplejava/src/main/java/com/example/chattutorialjava/ChannelActivity3.java

```diff
@@ -111,26 +109,28 @@ public void handleOnBackPressed() {
         typingHeader.setText(nobodyTyping);
 
         // Obtain a ChannelController
-        ChannelController channelController =
-                ChatDomain.instance().getUseCases()
-                        .getGetChannelController().invoke(cid)
-                        .execute().data();
-
-        // Observe typing users
-        channelController.getTyping().observe(this, users -> {
-            if (users.isEmpty()) {
-                typingHeader.setText(nobodyTyping);
-            } else {
-                StringBuilder typingText = new StringBuilder("typing: ");
-                for (int i = 0; i < users.size(); i++) {
-                    User user = users.get(i);
-                    if (i > 0) {
-                        typingText.append(", ");
+        ChatDomain.instance().getUseCases().getGetChannelController().invoke(cid).enqueue((result) -> {
+            ChannelController channelController = result.data();
+
+            runOnUiThread(() -> {
```

Comment:
Just FYI, after https://github.com/GetStream/stream-chat-android/pull/772 is done, we won't need this, as the callback will automatically put you back on the UI thread

Reply:
We should remember to update tutorials after that

Reply:
getGetChannelController doesn't event access the DB or API. So this will need another iteration till we fix that notification